src/movies.js

```diff
@@ -1,16 +1,179 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+
+function getAllDirectors(movies) {
+    let dirArr = movies.map((i) => {
+        return i.director
+    })
+    return dirArr
+}
+
+function cleanedArray(dirArr) {
+    let cleanArr
+
+}
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(movies) {
+    let steveDrama = movies.filter((i) => {
+        if (i.director === "Steven Spielberg" && i.genre.includes("Drama")) {
+            return i
+        }
+    }).length
+    return steveDrama
+}
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(movies) {
+    if (movies.length === 0) {
+        return 0
+    }
+    let numOfMovies = 0
+    let avgTot = movies.reduce((tot, e) => {
+        if (e.rate == undefined) {
+            numOfMovies += 1
+            return tot
+        } else {
+            numOfMovies += 1
+            return tot + e.rate
+        }
+    }, 0)
+
+    let avg = avgTot / numOfMovies
+    avg = avg.toFixed(2)
+    avg = Number(avg)
+    return avg
```

Comment:
the last 4 lines could be done in a single line, you are a pro, you can do this! :D